core/tee/tee_svc.c

```diff
@@ -313,17 +312,6 @@ static TEE_Result tee_svc_copy_param(struct tee_ta_session *sess,
 					  param_types);
 	}
 
-	if ((called_sess != NULL) &&
```

Comment:
Except that I've cleaned out some other kernel TA leftovers in this patch. I can try to break out what's possible of that cleanup in a separate patch if you like.